storage/wrapper.go

```diff
@@ -0,0 +1,168 @@
+package storage
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+type ctxKey string
+
+const (
+	DBCounterCtxKey ctxKey = "db_counter"
+)
+
+type DBCounter struct {
+	Reads  atomic.Int32
+	Writes atomic.Int32
+}
+
+// StorageWrapper is a wrapper around any OpenFGADatastore and is used to record metadata around how the underlying
+// implementation is being used.
+//
+// Currently, we are using it to record the number of reads and writes that are performed. Note that the values for the
+// reads and writes may not be accurate as we are only measuring calls that flow through the interface, and, such a call
+// may involve any number of database accesses under the hood.
```

Comment:
> the values for the  reads and writes may not be accurate as we are only measuring calls that flow through the interface, and, such a call may involve any **number of database accesses under the hood**

but this is exactly what i'd expect to see, especially if we are using one of the iterators... each call to `fetchResults()` should add 1 to the `read` counter

Reply:
I think that is too much work for too little value. We would have to add it to each implementation. As it is now, it gives a general idea... which there may be some value in:

Reply:
FWIW it definitely did not have that accuracy before either. In fact, it is more accurate now than before.